### PR TITLE
Partially revert "tests: Switch dockerhub registry images to promoted registry"

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

#### What type of PR is this?

Mitigate https://github.com/kubernetes/kubernetes/issues/99047 for now. This PR adds the ability for Windows jobs to point httpd registry to a different registry (i.e. https://github.com/kubernetes-sigs/windows-testing/pull/248)
/sig windows
/sig testing
/kind failing-test
/priority important-soon
/test pull-kubernetes-e2e-aks-engine-windows-contianerd

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster/images/conformance/Makefile [hunk 1]
```diff
@@ -54,6 +54,7 @@ endif
 	chmod a+rx ${TEMP_DIR}/e2e.test
 	chmod a+rx ${TEMP_DIR}/gorunner
 
+	docker buildx create --name img-builder --use
```
Comment (dims) [inline cluster/images/conformance/Makefile:57]: Is this a stray line or a necessary one @chewong ?

Comment (chewong) [reply cluster/images/conformance/Makefile:57]: We run `make quick-release` as part of pull-kubernetes-e2e-aks-engine-windows-contianerd and we kept getting the following error when building conformance image:
```
#12 [1/8] FROM docker.io/library/debian:stable-slim
#12 sha256:b98f3ae603cff0de4e4415d3bf6a561e4278041564c17f49727c53c13c046900
#12 resolve docker.io/library/debian:stable-slim 0.0s done
#12 ERROR: docker.io/library/debian:stable-slim not found
------
 > [internal] load metadata for docker.io/library/debian:stable-slim:
------
------
 > [1/8] FROM docker.io/library/debian:stable-slim:
------
error: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to build LLB: failed to load cache key: docker.io/library/debian:stable-slim not found
```
So I figure I might need to create a new buildx instance. Let me try again without this line.

Comment (chewong) [reply cluster/images/conformance/Makefile]: Without this line, the job failed. I am not sure if it's intermittent or not. Any idea?

**Comment by BenTheElder** [comment]
fyi @claudiubelu @spiffxp 
xref: https://github.com/kubernetes/kubernetes/pull/98964

**Comment by BenTheElder** [comment]
that test setup for AKS should probably set `KUBE_BUILD_CONFORMANCE=n` unless it's actually using the conformance image.

File: cluster/images/conformance/Makefile (hunk 1 above)
Comment (BenTheElder) [reply cluster/images/conformance/Makefile]: IMO CI and perhaps `make quick-release` should probably not be building this image, while it is a release artifact currently, it's not necessary for development workflows and building it in e2e etc. is mostly wasted resources.

I might consider disabling it except for `make release`, but even without that this job can disable it (see below)

But see also: https://github.com/kubernetes/kubernetes/pull/98672#issuecomment-771110925
This is actually a necessary fix (and probably isn't the complete fix), it should just be in another PR.

https://github.com/kubernetes/kubernetes/pull/99080 is establishing roughly the pattern we should need.

**Comment by chewong** [comment]
@dims, @claudiubelu PTAL. Thanks
